azkaban-common/src/main/java/azkaban/trigger/builtin/BasicTimeChecker.java

```diff
@@ -171,7 +171,13 @@ private long calculateNextCheckTime() {
         break;
       } else if (this.cronExecutionTime != null) {
         final Date nextDate = this.cronExecutionTime.getNextValidTimeAfter(date.toDate());
-        date = new DateTime(nextDate);
+        // It's possible to have a cron expression
+        // that will schedule a finite number of executions.
```

Comment:
don't quite understand this comment. Do you want to say "Some Cron Expressions possibly do not have follow-up occurrences"?

Reply:
@kunkun-tang That is correct. Would you want me to update that comment?